torch/distributed/elastic/rendezvous/etcd_rendezvous.py

```diff
@@ -0,0 +1,1259 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import datetime
+import json
+import logging
+import random
+import sys
+import threading
+import time
+from base64 import b64decode, b64encode
+from typing import Optional
+
+import etcd  # type: ignore[import]
```

Comment:
cc @malfet @seemethere is this dependency OK?

Reply:
I've talked to @seemethere, since this is a runtime dependency - for now we won't add the dependency to torch, but we will only add it to the test Docker image (so that we can reenable these tests). Users will have to `pip install etcd_client` if they want to use `EtcdRendezvous` but we will be adding a `StandaloneRendezvous` with no etcd runtime dependency.

Looks like we are doing something similar to `numpy` (users will have to manually install that dependency if they want to use an implementation in torch that depends on numpy) so we have precedence.